contracts/bonding/BondingManager.sol

```diff
@@ -1110,28 +1245,66 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
-            // across multiple transactions
-            require(_endRound.sub(_lastClaimRound) <= maxEarningsClaimsRounds, "too many rounds to claim through");
+            bool isTranscoder = _delegator == del.delegateAddress;
+
+            Transcoder storage transcoder = transcoders[del.delegateAddress];
 
-            for (uint256 i = startRound; i <= _endRound; i++) {
-                EarningsPool.Data storage earningsPool = transcoders[del.delegateAddress].earningsPoolPerRound[i];
+            // 'pendingFees()' and 'pendingStake()' are O(1) operations after LIP-36
+            if (startRound <= LIP_36_ROUND) {
+                // Cannot claim earnings for more than maxEarningsClaimsRounds before LIP-36
+                // This is a number to cause transactions to fail early if
+                // we know they will require too much gas to loop through all the necessary rounds to claim earnings
+                // The user should instead manually invoke `claimEarnings` to split up the claiming process
+                // across multiple transactions
+                uint256 endLoopRound = _endRound <= LIP_36_ROUND ? _endRound : LIP_36_ROUND;
+                require(endLoopRound.sub(_lastClaimRound) <= maxEarningsClaimsRounds, "too many rounds to claim through");
 
-                if (earningsPool.hasClaimableShares()) {
-                    (uint256 fees, uint256 rewards) = earningsPool.claimShare(currentBondedAmount, _delegator == del.delegateAddress);
+                while (startRound <= endLoopRound) {
+                    EarningsPool.Data storage earningsPool = transcoder.earningsPoolPerRound[startRound];
 
-                    currentFees = currentFees.add(fees);
-                    currentBondedAmount = currentBondedAmount.add(rewards);
+                    if (earningsPool.hasClaimableShares()) {
+                        (uint256 fees, uint256 rewards) = earningsPool.claimShare(currentBondedAmount, isTranscoder);
+                        currentFees = currentFees.add(fees);
+                        currentBondedAmount = currentBondedAmount.add(rewards);
+                    }
+
+                    startRound = startRound.add(1);
                 }
             }
-        }
 
+            // Check whether the endEarningsPool is initialised
+            // If it is not initialised set it's cumulative earnings factors (fees and rewards)
+            // This serves two purposes:
+            // - Initialise the correct values for the current claimEarnings 'epoch' to calculate cumulative earnings
+            // - Initialise the correct startEarningsPool for the next claimEarnings 'epoch' to calculate cumulative earnings
+            EarningsPool.Data storage endEarningsPool = transcoder.earningsPoolPerRound[_endRound];
+            if (endEarningsPool.cumulativeRewardFactor == 0) {
+                endEarningsPool.cumulativeRewardFactor = transcoder.earningsPoolPerRound[transcoder.lastRewardRound].cumulativeRewardFactor;
```

Comment:
fixed in 97980f7

Reply:
This storage update code looks unchanged?

Reply:
Oh sorry my bad.

These storage updates are necessary because they are not for a previous earnings pool but the start earnings pool when claiming. 

Essentially we copy over the values from the `lastRewardRound` and `lastFeeRound` to the earnings pool for the `lastClaimRound` because `lastRewardRound` and `lastFeeRound` is likely subject to change down the line so reading from those would yield different values.